plugins/modules/win_psrepository_copy.ps1

```diff
@@ -0,0 +1,234 @@
+#!powershell
+
+# Copyright: (c) 2020, Brian Scholer <@briantist>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#AnsibleRequires -CSharpUtil Ansible.Basic
+
+$spec = @{
+    supports_check_mode = $true
+    options = @{
+        source = @{
+            type = 'path'
+            default = '%LOCALAPPDATA%\Microsoft\Windows\PowerShell\PowerShellGet\PSRepositories.xml'
+        }
+        name = @{
+            type = 'list'
+            elements = 'str'
+            default = @(
+                '*'
+            )
+        }
+        exclude = @{
+            type = 'list'
+            elements = 'str'
+        }
+        profiles = @{
+            type = 'list'
+            elements = 'str'
+            default = @(
+                '*'
+            )
+        }
+        exclude_profiles = @{
+            type = 'list'
+            elements = 'str'
+            default = @(
+                'systemprofile'
+                'LocalService'
+                'NetworkService'
+            )
+        }
+    }
+}
+
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
+
+$module.Diff.changed = @{}
```

Comment:
I've always used `before`/`after` for the diff, does `changed` and `unchanged` also work?

Reply:
This was actually hard for me to tell.. there's not a lot of documentation about diff output. From what I could tell looking through a few other modules it seems like there's no actual defined structure/rules for what goes into `.Diff`, but I don't see the diff output unless both `--diff` and verbose are used. Is there supposed to be a different behavior and the reason is that I used the wrong fields? 

Happy to be wrong and update this if there's a better way.

Reply:
Yea it's not documented too well unfortunately. When implementing diff support in the past I've used 3 different keys

* before: String/dict/list containing the pre state
* after: String/dict/list containing the post state
* prepared: A string containing your own diff string

If you use before/after then Ansible itself handles the color/formatting itself which is great. An example would be `win_regedit`

![image](https://user-images.githubusercontent.com/8462645/105659818-69dd6100-5f15-11eb-9272-4d562a43651e.png)

You can see that Ansible adds the before/after difference similar to how git diff works. It even handled additions/removals to dicts/lists in a very nice way. Ultimately when I add diff support I have the before/after be an empty string to represent absent or a dict with each key being the relevant option for that object at that stage.

The prepared one is something I have used in the past but that was before I knew about before/after. It seems like `win_copy` is still using that. The downside of `prepared` is that the colors are now gone and you need to handle the before/after checks yourself. The plus side is you have full control over how the diff actually looks.

![image](https://user-images.githubusercontent.com/8462645/105660148-1ddeec00-5f16-11eb-88d2-f29cb158ad67.png)

When looking at the callback code that generates the diff text it does not seem like `changed`/`unchanged` is treated in any special way compared to `before`/`after` https://github.com/ansible/ansible/blob/484e4af4d0b0f8ba88f73c20f592bb107eb9396a/lib/ansible/plugins/callback/__init__.py#L168. I would recommend using before/after here instead but try it out and see what happens.

Reply:
Switched to `before`/`after`, seems to work just fine!